src/Microsoft.ML.TimeSeries/SsaAnomalyDetectionBase.cs

```diff
@@ -118,18 +118,21 @@ public SsaAnomalyDetectionBase(SsaArguments args, string name, IHostEnvironment
             ErrorFunction = args.ErrorFunction;
             ErrorFunc = ErrorFunctionUtils.GetErrorFunction(ErrorFunction);
             IsAdaptive = args.IsAdaptive;
-
             // Creating the master SSA model
             Model = new AdaptiveSingularSpectrumSequenceModeler(Host, args.InitialWindowSize, SeasonalWindowSize + 1, SeasonalWindowSize,
                 DiscountFactor, null, AdaptiveSingularSpectrumSequenceModeler.RankSelectionMethod.Exact, null, SeasonalWindowSize / 2, false, false);
+        }
 
+        public SsaAnomalyDetectionBase Fit(IDataView input)
```

Comment:
> public SsaAnomalyDetectionBase Fit(IDataView input) [](start = 7, length = 52)

this is a bad idea. Generally speaking, transformers should not have `Fit` methods